public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx

```diff
@@ -65,15 +65,7 @@ export const TimeSeriesPanel: React.FC<TimeSeriesPanelProps> = ({
       options={options}
     >
       {(config, alignedDataFrame) => {
-        alignedDataFrame.fields.forEach((field) => {
-          field.getLinks = getLinksSupplier(
-            alignedDataFrame,
-            field,
-            field.state!.scopedVars!,
-            replaceVariables,
-            timeZone
-          );
-        });
+        alignedDataFrame = regenerateLinksSupplier(alignedDataFrame, frames, replaceVariables, timeZone);
```

Comment:
Is there a way to skip this step when there are no data links or the aligned frame is the same as source frame?

Reply:
I just added a check to skip if no data links are present in the aligned df. the latter comparison gets tricky since the source frame is actually an array of dataframes.